distros/foxy/aruco-opencv/default.nix

```diff
@@ -16,7 +16,7 @@ buildRosPackage {
   buildType = "ament_cmake";
   buildInputs = [ ament-cmake ];
   checkInputs = [ ament-cmake-copyright ament-cmake-lint-cmake ament-cmake-uncrustify ament-cmake-xmllint ament-lint-auto ];
-  propagatedBuildInputs = [ aruco-opencv-msgs cv-bridge image-transport libyamlcpp python39Packages.img2pdf python3Packages.numpy python3Packages.opencv3 rclcpp rclcpp-components rclcpp-lifecycle tf2-geometry-msgs tf2-ros ];
+  propagatedBuildInputs = [ aruco-opencv-msgs cv-bridge image-transport libyamlcpp python39Packages.img2pdf python3Packages.numpy python3Packages.opencv4 rclcpp rclcpp-components rclcpp-lifecycle tf2-geometry-msgs tf2-ros ];
```

Comment:
I know that oyu didn't change this but shouldn't `python39Packages.img2pdf` be `python3Packages.img2pdf` to be consistent with the others?

Reply:
Probably. It's been 3.9 since img2pdf was added to rosdep: https://github.com/ros/rosdistro/pull/36241

A bit disappointing that the rosdistro maintainers didn't notice a problem, considering that it's different to every other Nix package name...

Reply:
A quick search reveals 16 uses of python39Packages in rosdep. I imagine these should all be changed.

Reply:
I submitted a PR https://github.com/ros/rosdistro/pull/38539

Reply:
Nice. Once the generated packages are updated I'll make a PR to bring Foxy in line.

In the meantime, this PR can probably still be merged.